python: make_transliterator type annotation too narrow: does not accept Transliterator instances in list

## Summary

`make_transliterator` has the following type signature:

```python
def make_transliterator(
    configs_or_recipe: list[TransliteratorConfig | TransliteratorIdentifier] | TransliterationRecipe,
) -> Callable[[str], str]:
```

However, the underlying `make_chained_transliterator` (which is called internally) accepts a broader type:

```python
def make_chained_transliterator(
    transliterator_configs: Iterable[TransliteratorConfig | TransliteratorIdentifier | Transliterator],
) -> Transliterator:
```

The difference is that `make_chained_transliterator` includes `Transliterator` as a valid item type, while `make_transliterator` does not.

## Impact

When a user passes a custom `Transliterator` instance directly in the list (which works fine at runtime), mypy reports an error:

```
error: List item N has incompatible type "MyTransliterator"; expected "Literal['archaic-hirakatas', ...] | tuple[Literal[...], dict[str, Any]]"  [list-item]
```

This forces callers to add `# type: ignore[list-item]` as a workaround.

## Expected behaviour

`make_transliterator` should accept `Transliterator` instances as list items, consistent with `make_chained_transliterator`:

```python
def make_transliterator(
    configs_or_recipe: list[TransliteratorConfig | TransliteratorIdentifier | Transliterator] | TransliterationRecipe,
) -> Callable[[str], str]:
```

## Reproduction

```python
from yosina import make_transliterator
from yosina.types import Transliterator
from yosina.chars import Char
from collections.abc import Iterable

class MyTransliterator:
    def __call__(self, input_chars: Iterable[Char]) -> Iterable[Char]:
        return input_chars

# mypy error: List item 1 has incompatible type "MyTransliterator"; expected "..."
fn = make_transliterator([("spaces", {}), MyTransliterator()])
```

## Environment

- yosina 1.1.3
- mypy 1.x

--- a/python/src/yosina/transliterator.py
+++ b/python/src/yosina/transliterator.py
@@ -8,12 +8,13 @@ from .chars import build_char_list, from_chars
 from .intrinsics import make_chained_transliterator
 from .recipes import TransliterationRecipe, build_transliterator_configs_from_recipe
 from .transliterators import TransliteratorConfig, TransliteratorIdentifier
+from .types import Transliterator
 
 __all__ = ["make_transliterator"]
 
 
 def make_transliterator(
-    configs_or_recipe: list[TransliteratorConfig | TransliteratorIdentifier] | TransliterationRecipe,
+    configs_or_recipe: list[TransliteratorConfig | TransliteratorIdentifier | Transliterator] | TransliterationRecipe,
 ) -> Callable[[str], str]:
     """Frontend convenience function to create a string-to-string transliterator.
 

